doc/src/devdocs/gc-mmtk.md

```diff
@@ -0,0 +1,43 @@
+# Julia + MMTk
+
+There has been quite a lot of effort to refactor the GC code inside Julia to support external GCs. The first step to enable using different GC algorithms for Julia was the design and implementation of a [GC interface](https://docs.google.com/document/d/1v0jtSrIpdEDNOxj5S9g1jPqSpuAkNWhr_T8ToFC9RLI/edit?usp=sharing). To drive that interface, we added support for building Julia with [MMTk](https://www.mmtk.io). MMTk is a memory management toolkit providing language implementers with a framework to implement flexible and performant GCs. The flexibility comes from the fact that it is possible to switch implementations fairly easily. MMTk supports state-of-the-art high-performance implementations that are continuously added and maintained in the core part of the framework. MMTk is under active development and has been used by other programming languages such as [Java](https://github.com/mmtk/mmtk-openjdk) and [Ruby](https://github.com/ruby/mmtk). To support a language, it is necessary to implement an *MMTk binding*, which contains the code that connects the language to [mmtk-core](https://github.com/mmtk/mmtk-core). The mmtk-julia binding can be found in [this repository](https://github.com/mmtk/mmtk-julia).
+
+> [!NOTE]
+> Using a different GC requires building Julia from source. It is not possible to switch implementations at runtime. To see what version of the GC is currently being used, run `versioninfo()` from the Julia REPL and it should show the version under `GC: ...`.
+
+## Building Julia with MMTk
+
+There are 3 different ways of building Julia with MMTk: building from source using a fixed release of the binding, checking out a custom version in the mmtk-julia [repository](https://github.com/mmtk/mmtk-julia) or using a precompiled binary from Julia's BinaryBuilder. The easiest way is to use the BinaryBuilder binary. First, to enable MMTk as a third-party GC, set the variable `WITH_THIRD_PARTY_GC` to `mmtk`. Then, simply set the variable `MMTK_PLAN` to one of the supported plans below and build Julia as usual.
+
+There are different configurations supported by the following variables, which can be set in a `Make.user` file or as an environment variable.
+
+| Variable      |       |        |
+|---------------|--------------|---------------|
+| `MMTK_PLAN`     | Immix        | StickyImmix   |
+| `MMTK_MOVING`   | 0            | 1             |
+| `MMTK_BUILD`    | release      | debug         |
```

Comment:
We're only instructing users to set `WITH_THIRD_PARTY_GC` and `MMTK_PLAN` in the `Make.user`, but then mention that there are other environment variables available (`MMTK_MOVING`, `MMTK_BUILD`).

This is a bit confusing, and it's possibly coming from the fact that we don't support moving yet.

I think we should mention that these variables can also be set, but at the time we're writing this, they will be no-ops, throw an error or something else because we don't support moving yet.